File: perl/src/tests/t_strategy.pl
#!/usr/bin/perl
#
# $Id:$
#
# perl test driver

use Atrshmlog;

$result = Atrshmlog::attach();

$id = Atrshmlog::get_strategy();

print "strategy : $id : \n";

$oldid = Atrshmlog::set_strategy(1);

print "strategy : $oldid : \n";

$id = Atrshmlog::get_strategy();

print "strategy : $id : \n";

$oldid = Atrshmlog::set_strategy(2);

print "strategy : $oldid : \n";

$id = Atrshmlog::get_strategy();

print "strategy : $id : \n";

$id = Atrshmlog::get_strategy_process();

print "strategy : $id : \n";

$oldid = Atrshmlog::set_strategy_process(4);

print "strategy : $oldid : \n";

$id = Atrshmlog::get_strategy_process();

print "strategy : $id : \n";

$oldid = Atrshmlog::set_strategy_process(0);

print "strategy : $oldid : \n";

$id = Atrshmlog::get_strategy_process();

print "strategy : $id : \n";


print " \n";

exit (0);

# end of main

File: python/src/tests/t_strategy.py
#!/usr/bin/python3
#
# Id:
#
# We test a bit of the atrshmlog here.
#
# This is for the first starter, so only the basic things.

import sys
import atrshmlog

result = atrshmlog.attach()

id = atrshmlog.get_strategy()

print('strategy : ' + str(id) + ' : ')

oldid = atrshmlog.set_strategy(1)

print('strategy : ' + str(oldid) + ' : ')

id = atrshmlog.get_strategy()

print('strategy : ' + str(id) + ' : ')

oldid = atrshmlog.set_strategy(2)

print('strategy : ' + str(oldid) + ' : ')

id = atrshmlog.get_strategy()

print('strategy : ' + str(id) + ' : ')

id = atrshmlog.get_strategy_process()

print('strategy : ' + str(id) + ' : ')

oldid = atrshmlog.set_strategy_process(4)

print('strategy : ' + str(oldid) + ' : ')

id = atrshmlog.get_strategy_process()

print('strategy : ' + str(id) + ' : ')

oldid = atrshmlog.set_strategy_process(0)

print('strategy : ' + str(oldid) + ' : ')

id = atrshmlog.get_strategy_process()

print('strategy : ' + str(id) + ' : ')

print (' ')

exit(0)

# end of test

